gammapy/region/core.py

```diff
@@ -0,0 +1,231 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+import abc
+from astropy.extern import six
+
+
+__all__ = [
+    'Region',
+    'PixRegion',
+    'SkyRegion',
+    'RegionList',
+]
+
+@six.add_metaclass(abc.ABCMeta)
+class Region(object):
+    """
+    Base class for all regions.
+    """
+
+    def intersection(self, other):
+        """
+        Returns a region representing the intersection of this region with
+        ``other``.
+        """
+        raise NotImplementedError("")
+
+    def symmetric_difference(self, other):
+        """
+        Returns the union of the two regions minus any areas contained in the
+        intersection of the two regions.
+        """
+        raise NotImplementedError("")
+
+    def union(self, other):
+        """
+        Returns a region representing the union of this region with ``other``.
+        """
+        raise NotImplementedError("")
+
+
+@six.add_metaclass(abc.ABCMeta)
+class PixRegion(Region):
+    """
+    Base class for all regions defined in pixel coordinates
+    """
+
+    def __contains__(self, pixcoord):
+        """
+        Checks whether a position or positions fall inside the region.
+
+        Parameters
+        ----------
+        pixcoord : tuple
+            The position or positions to check, as a tuple of scalars or
+            arrays. In future this could also be a `PixCoord` instance.
+        """
+        raise NotImplementedError("")
+
+    def area(self):
+        """
+        Returns the area of the region as a `~astropy.units.Quantity`.
+        """
+        raise NotImplementedError("")
+
+    def to_sky(self, wcs, mode='local', tolerance=None):
+        """
+        Returns a region defined in sky coordinates.
+
+        Parameters
+        ----------
+
+        wcs : `~astropy.wcs.WCS` instance
+            The world coordinate system transformation to assume
+
+        mode : str
```

Comment:
I think we have no intention of supporting "affine" and "full" mode, right?
Then remove the "mode" and "tolerance" arguments here.